### PR TITLE
Make the spec look more like a spec

### LINKED ISSUE #353: More finely chunk IDL so there are places to write prose
Right now, the spec's IDL code blocks are really big, so putting prose way down at the bottom for something in the middle isn't really practical.

I intend (but haven't yet started) to do this, and move any comments out to text while I'm at it.

### PR DESCRIPTION
**Pull request by kainino0x**
- Chunk the IDL into small parts
- Move prose from comments into the newly created spaces
- Re-order some things for consistency and readability

Fixes #353

***
<a href="https://pr-preview.s3.amazonaws.com/kainino0x/gpuweb/pull/365.html" title="Last updated on Jul 12, 2019, 8:52 PM UTC (58af94d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/365/918c28b...kainino0x:58af94d.html" title="Last updated on Jul 12, 2019, 8:52 PM UTC (58af94d)">Diff</a>

### CODE REVIEW

**Comment by kainino0x** [comment]
This PR doesn't change anything substantive (except for the changes in the first commit, which is #364), so please review the overall structure in the non-diff rendered version.

**Comment by kainino0x** [comment]
(In particular, the way that prose is inserted between code blocks. No need to review the structure in detail yet - I'm sure it will continue changing.)

FYI, I was thinking of opening another PR which splits some of the interfaces into many partial interfaces, so that each function can be defined in a more useful place (e.g. createBuffer in the Buffers section). Haven't settled on that idea yet though.

File: spec/index.bs [hunk 1]
```diff
@@ -80,53 +119,134 @@ interface GPUBufferUsage {
...
-interface GPUBuffer : GPUObjectBase {
-    Promise<ArrayBuffer> mapReadAsync();
-    Promise<ArrayBuffer> mapWriteAsync();
-    void unmap();
+<script type=idl>
+typedef sequence<any> GPUMappedBuffer;
+</script>
+
+{{GPUMappedBuffer}} must be a sequence of 2 elements of types {{GPUBuffer}} and
+{{ArrayBuffer}}, respectively. Otherwise, an exception is thrown.
```
Comment (Kangz) [inline spec/index.bs]: `GPUMappedBuffer` is a type returned by the implementation, not an argument passed by the application, so I don't think we need to specify the exception bit.

Comment (kainino0x) [reply spec/index.bs]: Oh, right. Thanks!

**Comment by kainino0x** [comment]
Thanks for the reviews!

**Pull request merged**